src/audio_decoder.cpp

```diff
@@ -105,19 +104,18 @@ class WMAUnsupportedFormatDecoder : public AudioDecoder {
 		error_message = std::string("WMA audio files are not supported. Reinstall the\n") +
 			"game and don't convert them when asked by Windows!\n";
 	}
-	bool Open(FILE*) override { return false; }
+	bool Open(std::shared_ptr<FileFinder::istream> stream) override { return false; }
 	bool IsFinished() const override { return true; }
 	void GetFormat(int&, Format&, int&) const override {}
 private:
 	int FillBuffer(uint8_t*, int) override { return -1; };
 };
 const char wma_magic[] = { (char)0x30, (char)0x26, (char)0xB2, (char)0x75 };
 
-std::unique_ptr<AudioDecoder> AudioDecoder::Create(FILE* file, const std::string& filename, bool resample) {
+std::unique_ptr<AudioDecoder> AudioDecoder::Create(std::shared_ptr<FileFinder::istream> stream, const std::string& filename, bool resample) {
```

Comment:
Is `AudioDecoder` taking ownership of the stream? If so I would just pass `istream&` and forget the `shared_ptr`.